tools/odbc/api_info.cpp

```diff
@@ -216,9 +217,6 @@ SQLRETURN ApiInfo::GetFunctions(SQLHDBC connection_handle, SQLUSMALLINT function
 }
 
 SQLRETURN ApiInfo::GetFunctions30(SQLHDBC connection_handle, SQLUSMALLINT function_id, SQLUSMALLINT *supported_ptr) {
-	if (function_id != SQL_API_ODBC3_ALL_FUNCTIONS) {
-		return SQL_ERROR;
```

Comment:
Can we remove this, or do we need to return a different error? Or do we throw it somewhere else now?

Reply:
Because `GetFunctions30` is only called if the `function_id` is `SQL_API_ODBC3_ALL_FUNCTIONS`, so it seems unnecessary to check for it twice:
```cpp
SQLRETURN SQL_API SQLGetFunctions(SQLHDBC connection_handle, SQLUSMALLINT function_id, SQLUSMALLINT *supported_ptr) {
	if (function_id == SQL_API_ODBC3_ALL_FUNCTIONS) {
		return ApiInfo::GetFunctions30(connection_handle, function_id, supported_ptr);
	} else {
		return ApiInfo::GetFunctions(connection_handle, function_id, supported_ptr);
	}
}
```